### PR TITLE
Use new syntax for xcom pull in example dags

### PR DESCRIPTION
**Pull request by hsnprsd**
Part of https://github.com/apache/airflow/issues/10285.

### CODE REVIEW

File: airflow/example_dags/example_xcom.py [hunk 1]
```diff
@@ -43,24 +44,14 @@ def push_by_returning(**kwargs):
...
 
-    # get value_2
-    pulled_value_2 = ti.xcom_pull(task_ids='push_by_returning')
-    if pulled_value_2 != value_2:
-        raise ValueError(f'The two values differ {pulled_value_2} and {value_2}')
-
-    # get both value_1 and value_2
-    pulled_value_1, pulled_value_2 = ti.xcom_pull(key=None, task_ids=['push', 'push_by_returning'])
-    if pulled_value_1 != value_1:
-        raise ValueError(f'The two values differ {pulled_value_1} and {value_1}')
```
Comment (turbaszek) [inline airflow/example_dags/example_xcom.py:63]: I would still keep an example of accessing `ti` from `kwargs` for example:
```
# check pulled value 1
...
# same using explicit xcom pull
from_xcom = ti.xcom_pull(key=None, task_ids='push')
if from_xcom != pulled_value_1:
    raise ValueError(f'The two values differ {pulled_value_1} and {value_1}
```
WDYT?

File: airflow/example_dags/example_xcom.py [hunk 2]
```diff
@@ -43,24 +44,14 @@ def push_by_returning(**kwargs):
     return value_2
 
 
-def puller(**kwargs):
+def puller(pulled_value_1, pulled_value_2, **kwargs):
```
Comment (turbaszek) [inline airflow/example_dags/example_xcom.py:47]:
```suggestion
def puller(pulled_value_1, pulled_value_2):
```
We can probably drop the `kwargs`

File: airflow/example_dags/tutorial_etl_dag.py [hunk 3]
```diff
@@ -61,33 +61,35 @@
...
-        total_value_json_string = json.dumps(total_value)
-        ti.xcom_push('total_order_value', total_value_json_string)
+        total_order_value_string = json.dumps(
+            {
+                "total_order_value": total_order_value,
+            }
+        )
+
+        ti = kwargs['ti']
+        ti.xcom_push('total_order_value', total_order_value_string)
```
Comment (turbaszek) [inline airflow/example_dags/tutorial_etl_dag.py]: This looks like unrelated change

File: airflow/example_dags/tutorial_etl_dag.py [hunk 4]
```diff
@@ -108,6 +110,7 @@ def load(**kwargs):
     transform_task = PythonOperator(
         task_id='transform',
         python_callable=transform,
+        op_args=[extract_task.output['order_data']],
```
Comment (turbaszek) [inline airflow/example_dags/tutorial_etl_dag.py]: If I'm not mistaken we would like to keep the old way in this example because we use it to compare the old vs. new approach. CC @vikramkoka ?

Comment (ashb) [reply airflow/example_dags/tutorial_etl_dag.py]: Correct. This is dag is to show the classic way, and tutorial_taskflow_api_etl.py is the companion showing the new way.

**Review by ashb — CHANGES_REQUESTED** [review]
Please revert all changes to airflow/example_dags/tutorial_etl_dag.py

File: airflow/example_dags/tutorial_etl_dag.py (hunk 4 above)
Comment (vikramkoka) [reply airflow/example_dags/tutorial_etl_dag.py]: Yes, that is correct. This was intended to show the "classic way".

File: airflow/example_dags/example_xcom.py (hunk 1 above)
Comment (hsnprsd) [reply airflow/example_dags/example_xcom.py:63]: I do agree. Will fix it.

File: airflow/example_dags/example_xcom.py (hunk 2 above)
Comment (hsnprsd) [reply airflow/example_dags/example_xcom.py:47]: I think I need that for getting the ti as we want to put an example using the old syntax and comparing it to new one like you mentioned in your previous comment.

File: airflow/example_dags/tutorial_etl_dag.py (hunk 3 above)
Comment (hsnprsd) [reply airflow/example_dags/tutorial_etl_dag.py]: I reverted all changes in this file.

File: airflow/example_dags/example_xcom.py [hunk 5]
```diff
@@ -81,6 +78,10 @@ def puller(**kwargs):
     task_id='puller',
     dag=dag,
     python_callable=puller,
+    op_args=[
+        push1.output['value_from_pusher 1'],
```
Comment (turbaszek) [inline airflow/example_dags/example_xcom.py]:
```suggestion
        push1.output['value from pusher 1'],
```
Otherwise the DAG is failing:
```
*** Reading local file: /root/airflow/logs/example_xcom/puller/2020-11-18T10:28:58.185921+00:00/1.log
[2020-11-18 10:29:04,239] {taskinstance.py:827} INFO - Dependencies all met for <TaskInstance: example_xcom.puller 2020-11-18T10:28:58.185921+00:00 [queued]>
[2020-11-18 10:29:04,285] {taskinstance.py:827} INFO - Dependencies all met for <TaskInstance: example_xcom.puller 2020-11-18T10:28:58.185921+00:00 [queued]>
[2020-11-18 10:29:04,287] {taskinstance.py:1018} INFO - 
--------------------------------------------------------------------------------
[2020-11-18 10:29:04,289] {taskinstance.py:1019} INFO - Starting attempt 1 of 1
[2020-11-18 10:29:04,290] {taskinstance.py:1020} INFO - 
--------------------------------------------------------------------------------
[2020-11-18 10:29:04,306] {taskinstance.py:1039} INFO - Executing <Task(PythonOperator): puller> on 2020-11-18T10:28:58.185921+00:00
[2020-11-18 10:29:04,313] {standard_task_runner.py:50} INFO - Started process 37596 to run task
[2020-11-18 10:29:04,327] {standard_task_runner.py:74} INFO - Running: ['airflow', 'tasks', 'run', 'example_xcom', 'puller', '2020-11-18T10:28:58.185921+00:00', '--job-id', '8', '--pool', 'default_pool', '--raw', '--subdir', '/opt/airflow/airflow/example_dags/example_xcom.py', '--cfg-path', '/tmp/tmpdql6__s7']
[2020-11-18 10:29:04,331] {standard_task_runner.py:75} INFO - Job 8: Subtask puller
[2020-11-18 10:29:04,477] {logging_mixin.py:103} INFO - Running <TaskInstance: example_xcom.puller 2020-11-18T10:28:58.185921+00:00 [running]> on host 557abba307a4
[2020-11-18 10:29:04,540] {taskinstance.py:1402} ERROR - XComArg result from push at example_xcom with key="value_from_pusher 1"" is not found!
Traceback (most recent call last):
  File "/opt/airflow/airflow/models/taskinstance.py", line 1087, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/opt/airflow/airflow/models/taskinstance.py", line 1224, in _prepare_and_execute_task_with_callbacks
    self.render_templates(context=context)
  File "/opt/airflow/airflow/models/taskinstance.py", line 1690, in render_templates
    self.task.render_template_fields(context)
  File "/opt/airflow/airflow/models/baseoperator.py", line 857, in render_template_fields
    self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
  File "/opt/airflow/airflow/models/baseoperator.py", line 870, in _do_render_template_fields
    rendered_content = self.render_template(content, context, jinja_env, seen_oids)
  File "/opt/airflow/airflow/models/baseoperator.py", line 921, in render_template
    return [self.render_template(element, context, jinja_env) for element in content]
  File "/opt/airflow/airflow/models/baseoperator.py", line 921, in <listcomp>
    return [self.render_template(element, context, jinja_env) for element in content]
  File "/opt/airflow/airflow/models/baseoperator.py", line 909, in render_template
    return content.resolve(context)
  File "/opt/airflow/airflow/models/xcom_arg.py", line 133, in resolve
    raise AirflowException(
airflow.exceptions.AirflowException: XComArg result from push at example_xcom with key="value_from_pusher 1"" is not found!
[2020-11-18 10:29:04,554] {taskinstance.py:1439} INFO - Marking task as FAILED. dag_id=example_xcom, task_id=puller, execution_date=20201118T102858, start_date=20201118T102904, end_date=20201118T102904
[2020-11-18 10:29:04,615] {local_task_job.py:118} INFO - Task exited with return code 1
```

**Review by turbaszek — APPROVED** [review]

**Comment by ashb** [comment]
The linked issue talked about updating

> airflow/example_dags/* to use .output instead of jinja whenever it is possible

But this PR is now only changing a single file. Was there some discussion I missed? I expected all the example dags other than the `tutorial_etl_dag` to be updated to the new syntax.

**Comment by hsnprsd** [comment]
@ashb I can't find any other example dag using xcom.

There was three files using xcoms.
`example_xcom.py`: which I fixed it.
`tutorial_etl_dag.py`: which we are gonna use the old syntax here as you told.
`example_xcomargs.py`: which is already using the new syntax and needs no change.

Correct me if I'm missing something.

**Comment by ashb** [comment]
Oh! I just somehow assumed there would be more cases.

I guess the other ones are in the providers.

Here is a complete list of every example dag that has the string `xcom_pull` in it:

```
airflow/providers/amazon/aws/example_dags/example_emr_job_flow_automatic_steps.py
airflow/providers/amazon/aws/example_dags/example_google_api_to_s3_transfer_advanced.py
airflow/providers/amazon/aws/example_dags/example_glacier_to_gcs.py
airflow/providers/jenkins/example_dags/example_jenkins_job_trigger.py
airflow/providers/amazon/aws/example_dags/example_emr_job_flow_manual_steps.py
airflow/providers/salesforce/example_dags/example_tableau_refresh_workbook.py
airflow/providers/docker/example_dags/example_docker_copy_data.py
airflow/providers/apache/kylin/example_dags/example_kylin_dag.py
airflow/providers/google/marketing_platform/example_dags/example_search_ads.py
airflow/providers/google/marketing_platform/example_dags/example_campaign_manager.py
airflow/providers/google/suite/example_dags/example_sheets.py
airflow/providers/google/cloud/example_dags/example_datastore.py
airflow/providers/google/suite/example_dags/example_gcs_to_sheets.py
airflow/providers/google/cloud/example_dags/example_bigquery_dts.py
airflow/providers/google/cloud/example_dags/example_cloud_storage_transfer_service_aws.py
airflow/providers/google/cloud/example_dags/example_automl_nl_text_classification.py
airflow/providers/google/cloud/example_dags/example_dataproc.py
airflow/providers/google/cloud/example_dags/example_translate.py
airflow/providers/google/cloud/example_dags/example_video_intelligence.py
airflow/providers/google/cloud/example_dags/example_automl_tables.py
airflow/providers/google/cloud/example_dags/example_cloud_storage_transfer_service_gcp.py
airflow/providers/google/cloud/example_dags/example_dlp.py
airflow/providers/google/cloud/example_dags/example_datacatalog.py
airflow/providers/google/cloud/example_dags/example_dataflow.py
airflow/providers/google/cloud/example_dags/example_bigquery_operations.py
airflow/providers/google/cloud/example_dags/example_automl_video_intelligence_classification.py
airflow/providers/google/cloud/example_dags/example_vision.py
airflow/providers/google/cloud/example_dags/example_kubernetes_engine.py
airflow/providers/google/cloud/example_dags/example_cloud_memorystore.py
airflow/providers/google/cloud/example_dags/example_gcs.py
airflow/providers/google/cloud/example_dags/example_automl_nl_text_sentiment.py
airflow/providers/google/cloud/example_dags/example_cloud_build.py
airflow/providers/google/cloud/example_dags/example_mlengine.py
airflow/providers/google/cloud/example_dags/example_bigquery_queries.py
airflow/providers/google/cloud/example_dags/example_automl_nl_text_extraction.py
airflow/providers/google/cloud/example_dags/example_pubsub.py
airflow/providers/google/cloud/example_dags/example_automl_translation.py
airflow/providers/google/cloud/example_dags/example_automl_vision_classification.py
airflow/providers/google/cloud/example_dags/example_automl_vision_object_detection.py
airflow/providers/google/cloud/example_dags/example_automl_video_intelligence_tracking.py
airflow/providers/google/marketing_platform/example_dags/example_display_video.py
airflow/providers/google/cloud/example_dags/example_natural_language.py
airflow/providers/google/cloud/example_dags/example_cloud_sql.py
airflow/providers/cncf/kubernetes/example_dags/example_spark_kubernetes.py
airflow/providers/cncf/kubernetes/example_dags/example_kubernetes.py
airflow/example_dags/example_xcom.py
```

Do you have time to update all of these too? That would be ace if so!

**Comment by turbaszek** [comment]
> Do you have time to update all of these too?

@ashb  I would say it's not only about time but also about access to resources. The DAGs should be run at least once to make sure the changes work. In case of google we are going to rewrite operators to use google api 2.0 that's when we will be able to refactor and test everything - would it make sense? (CC @michalslowikowski00 )

**Comment by ashb** [comment]
> In case of google we are going to rewrite operators to use google api 2.0

I'd rather not wait until after 2.0 for this -- having everything using the shiny new syntax for 2.0 release would make me happier.

**Comment by hsnprsd** [comment]
@ashb I can change the files you mentioned but I don't think I can test them as I don't have access to the providers like AWS or GC. Is it OK for me to change and someone else to test?

**Comment by ashb** [comment]
> @ashb I can change the files you mentioned but I don't think I can test them as I don't have access to the providers like AWS or GC. Is it OK for me to change and someone else to test?

Sounds good!
